Fix: Improve autoconfig logging (fixes #5119)

IanVS opened the pull request:
This also adds additional tests to bring the coverage back up, and fixes some bugs which were found during testing.

ilyavolodin commented on lib/util/source-code-util.js:
```diff
@@ -25,14 +29,19 @@ var assign = require("object-assign"),
+    }
     var sourceCode = eslint.getSourceCode();
-    return sourceCode;
+    return [sourceCode, null];
```
Wouldn't it be easier to throw an error if you can't parse the file and run the code inside try/catch statement?

IanVS replied:
It __would__ be great to throw an error, but for the life of me I can't figure out how to get inquirer to handle or bubble up errors.  Instead, it just swallows them silently.  :-/

nzakas replied:
You need to use a try-catch whjen you call this function and then pass the error to cb().

IanVS replied:
Cool, thanks.  I'll give that a try.

gyandeeps commented:
@IanVS Can we plz use this PR to increase code coverage for this functionality? 
Initially with this functionality our coverage got drooped. It takes a big effort later on to pick up code coverage where a single person has to go all out on the whole code base (Which I did recently). Specially the branch coverage.

IanVS commented:
Yes, I planned to add some more tests at some point anyway.  Is there a way to find the areas of code that are not covered?  Do I just need to go read the istanbul docs?

gyandeeps commented:
You can open up index file inside eslint\coverage\lcov-report folder on your machine. That where istanbul creates the code coverage. Its an html file so it will give you all the information.

ilyavolodin commented:
To be more specific, once you run `npm test` locally, istanbul will create `/coverage/` folder with all coverage information you might need. Just open it up in the browser and you should see full code coverage and can dig into specific files/functions.

IanVS commented:
OK, with the latest change the whole process will error if a parsing error is encountered (thanks to @nzakas explaining how to throw an error).  I think this is better than creating a config as long as at least __one__ file was parsable (which is what would have happened before).

I still need to add some more tests to increase coverage back up.

IanVS commented:
I've brought the code coverage back up, and sure enough I found a few bugs. 

This also adds support for the new rules with schemas nested under a `meta` property.

I believe I've addressed all of the comments so far.

gyandeeps commented:
Excellent work thr @IanVS ... Thanks a lot. :)

nzakas commented:
@ilyavolodin want to double check the meta stuff?

ilyavolodin commented:
LGTM
